### DISCUSSION TITLE
AbpHelper vNext

### DISCUSSION THREAD

**Discussion by gdlcf88** [EasyAbp/AbpHelper.CLI#195] [Ideas]
The new AbpHelper provides more generation features and the following configurations persistence for each solution.

* AggregateRoot
  * BaseInfo
    * Name
    * IsMultiTenancy: true
    * AuditType: FullAudited, Audited, CreationAudited, None
  * Application
    * AppServiceType: CrudAppService, AppService
    * IsReadonlyService: false
    * CreateUpdateDtoType: Separated, Merged
    * Permission:
    *   * HasBasicPermission: true
    *   * HasManagePermission: false
  * Domain
    * HasEntityInterface: false
    * HasEntityEto: false
    * DomainService
    *   * HasDomainService: false
    *   * HandleEntityDeletion: false
    * UniqueCheck: PersistenceConstraint, DomainServiceCheckWithLock, DomainServiceCheck
    * HasEntityCache: false
    * TrackEntityHistory: false
  * UI
    * HasUserInterface: true
    * UiFramework: Mvc
  * HttpApi
    * HasManualController: true
  * Test
    * HasTest: true
  * List\<Property\>
  * List\<Entity\>
  * List\<UniqueConstraint\>

* Entity
  * BaseInfo
    * Name
    * IsMultiTenancy: true
    * AuditType: FullAudited, Audited, CreationAudited, None
  * List\<Property\>
  * List\<Entity\>
  * List\<UniqueConstraint\>

* Property
  * BaseInfo
    * Name
    * IsKey: false (true as default if Name == "Id")
    * AuditingDisabled: false
  * TypeInfo
    * Type: String, Guid, Boolean, DateTime, Decimal, Double, Byte, Short, Int, Long, Enum, ValueObject
    * IsNullable: false
    * GetterType: Public, GetterMethod, Protected
    * SetterType: Protected, SetterMethod, Public
    * MinValue: null
    * MaxValue: null
    * DecimalPrecision: null, e.g. 20
    * DecimalScale: null, e.g. 8
  * Documentation
    * DocumentationSummary
    * DocumentationRemark
  * Usability
    * EntityDtoIncluded: true
    * EntityEtoIncluded: true
    * CreationIncluded: true
    * UpdateIncluded: true
    * UiVisible: true
    * UiShowInList: true
    * GetListFilterIncluded: true
    * HasGetByMethod: true, effective when IsUnique == true
  * EntityDataSynchronization
    * DataSource: LocalEventBus, DistributedEventBus, Custom
    * TargetEntityName: e.g. Man
    * TargetEntityPropertyName: e.g. Name
    * TargetEntityKeyPropertyNames: e.g. RegionId,GroupId,Code
    * LocalForeignKeyPropertyNames: e.g. ManRegionId,ManGroupId,ManCode

* UniqueConstraint
  * PropertyNames: List\<string\>

* TemplateVariables
  * EntityDtoName: {{entityName}}Dto
  * CreateUpdateDtoName: CreateUpdate{{entityName}}Dto
  * CreateDtoName: Create{{entityName}}Dto
  * UpdateDtoName: Update{{entityName}}Dto
  * GetListInputName: {{entityName}}GetListInput

* Other Generations
  * Exception
  * Localization
    * List, Create, Edit, Delete
    * Reordering
  * EF Migration

**Comment by gdlcf88** [comment]
@blackWins - Do you want to help?

**Reply by blackWins** [reply]
I'm still at the entry-level stage, and the code quality may not be very high. I can try coding

**Reply by gdlcf88** [reply]
Code doesn't matter. Contact me on my WeChat ID `lanxiajiao` if you are interested. We can talk about the details. 😃
